docs/specifications/linking-and-embedding-specification.md

```diff
@@ -0,0 +1,364 @@
+
+The rules governing the path-based linking are as follows:
+
+- Pointing to a note is done via its workspace rooted path, e.g. `[[some/path/here.md]]`
+- Links can also be written without the file extensions, like `[[some/path/here]]`
+- In case there's conflicts due to lack of file extensions (e.g. `here.md` and `here.doc` in the same path), link reference definitions are used to disambiguate (e.g. `[some-file]: some-file.md`) OR the file extension is added into the link itself OR an error is shown to the user.
+- Pointing to a heading within a note is done based on id, e.g. `[[foo/bar#my-title]]`
+- Pointing to an arbitrary anchor within a note is done also based on id, e.g. `[[foo/bar#my-anchor]]`
+
+##### Link Labels
+
+All wikilinks can have an optional label assigned to them. The assignment is done by separating the link target and the label with a pipe-character: `[[Some Note#Some Anchor|Some Label]]`
+
+##### Linking to aliased notes
+
+As explained in the [aliasing chapter](#aliases), a note can define aliases to itself. These aliases are used only to look up the correct title / path for the link, and they are not recorded into the links themselves. 
+
+##### Using Tags
+
+Tags can be referred (i.e. linked to) using the normal hashtag-syntax anywhere in the text (e.g. `It's raining #cats and #dogs.`). Additionally, tags can be referred to using the front matter in the following way:
+
+  ```yaml
+  ---
+  tags: cats, dogs
+  ---
+  ```
+
+Tags in the front matter may contain the `#`-character (i.e. `tags: #cats, #dogs`), they are interpreted in the same way as the tags not containing the character (i.e. `tags: cats, #dogs` is equal and valid).
+
+If a tag is file-based and it contains `/`-characters, this denotes a subfolder in the `tags/`-folder. Tags ending with `/` character act the same as a tag without that character. By default, tags are abstract and can be converted into files by the user.
+
+### Linking Local Non-Markdown Files
+
+Any file in the workspace (e.g. excel-files) can be linked using one of the following 1) normal markdown link syntax, 2) path-based wikilink syntax, or 3) word-based wikilinks with arbitrary text using the link reference definitions.
+
+### Linking External Locations
+
+Linking external locations (e.g. web pages) is always done using one of the normal markdown link syntaxes.
+
+## Embedding
+
+Embedding differs from linking mainly when inspecting the workspace after it has been built e.g. into html format. While linking just adds a link into the page, embedding embeds the whole target into the location where it is referred to.
+
+Embedding is done with the same syntax as linking with one exception, adding an exclamation mark (`!`) in front of the link. Therefore `[[Some Page]]` adds just a link, while `![[Some Page]]` replaces that reference with the actual page. 
```

Comment:
I like this, as it mirrors the markdown image syntax, which is a sort of an embed!

Reply:
I'm wondering if this can cause some conflicts in certain languages: for example spanish starts exclamation sentences with `¡` (and ends with `!`) and so a link right after the start of a sentence would be interpreted as an embed. (Really not sure but might be worth researching)